src/index.ts

```diff
@@ -216,20 +222,55 @@ export class PriceFeed {
 
   /**
-   * Get the "unchecked" previous price with Trading status, along with the timestamp at which it was generated.
+   * Get the latest available price, along with the timestamp when it was generated.
+   * This function returns the same price as `getCurrentPrice` in the case where a price was available
+   * at the time this `PriceFeed` was published (`publish_time`). However, if a price was not available
+   * at that time, this function returns the price from the latest time at which the price was available.
+   * The returned price can be from arbitrarily far in the past; this function makes no guarantees that
+   * the returned price is recent or useful for any particular application.
    *
-   * @returns a struct containing the previous price, confidence interval, and the exponent for
-   * both numbers along with the timestamp that the price was generated.
+   * Users of this function should check the returned timestamp to ensure that the returned price is
+   * sufficiently recent for their application. If you are considering using this function, it may be
+   * safer / easier to use either `getCurrentPrice` or `getLatestAvailablePriceWithinDuration`.
    *
-   * WARNING:
-   * We make no guarantees about the unchecked price and confidence returned by
-   * this function: it could differ significantly from the current price.
-   * We strongly encourage you to use `get_current_price` instead.
+   * @returns a struct containing the latest available price, confidence interval, and the exponent for
+   * both numbers along with the timestamp when that price was generated.
    */
-  getPrevPriceUnchecked(): [Price, UnixTimestamp] {
+  getLatestAvailablePriceUnchecked(): [Price, UnixTimestamp] {
+    // If the price status is Trading then it's the latest price
+    // with the Trading status.
+    if (this.status === PriceStatus.Trading) {
+      return [new Price(this.conf, this.expo, this.price), this.publishTime];
+    }
+
     return [
       new Price(this.prevConf, this.expo, this.prevPrice),
       this.prevPublishTime,
     ];
   }
+
+  /**
+   * Get the latest price as long as it was updated within `duration` seconds of the current time.
+   * This function is a sanity-checked version of `getLatestAvailablePriceUnchecked` which is useful in
+   * applications that require a sufficiently-recent price. Returns `undefined` if the price wasn't
+   * updated sufficiently recently.
+   *
+   * @param duration return a price as long as it has been updated within this number of seconds
+   * @returns a struct containing the latest available price, confidence interval and the exponent for
+   * both numbers, or `undefined` if no price update occurred within `duration` seconds of the current time.
+   */
+  getLatestAvailablePriceWithinDuration(duration: DurationInSeconds): Price | undefined {
+    const [price, timestamp] = this.getLatestAvailablePriceUnchecked();
+
+    const currentTime: UnixTimestamp = Math.floor(Date.now() / 1000);
```

Comment:
FYI we may want to make the current time a parameter to this function so it's a pure function. It's easier to test, and it lets users choose their clock.

This is fine for now, but something to consider in the future.

Reply:
Currently all sdks use current time as a system time (or block time) or do not use it at all.
As I'm gonna change all sdks let's decide on it so I change everything once :D